public/app/features/dashboard/state/initDashboard.ts

```diff
@@ -181,7 +180,17 @@ export function initDashboard(args: InitDashboardArgs): ThunkResult<void> {
     // template values service needs to initialize completely before
     // the rest of the dashboard can load
     try {
-      await variableSrv.init(dashboard);
+      if (!getConfig().featureToggles.newVariables) {
+        await variableSrv.init(dashboard);
+      }
+      if (getConfig().featureToggles.newVariables) {
+        const list =
+          dashboard.variables.list.length > 0
+            ? dashboard.variables.list
+            : dashboard.templating.list.filter(v => v.type === 'query');
```

Comment:
Great question!! We you this just for testing purposes so existing dashboards are not shown without variables. This will be moved to a proper dashboard migration when all types are moved.